### DISCUSSION TITLE
messages pending

### DISCUSSION THREAD

**Discussion by julianorib** [postalserver/postal#2207] [Help with using Postal]
Hello
I've been using postal 2.
is working almost normally.

there is an issue where some messages are in pending status.
if I filter them and click on "release messages" they are sent.

I would like help trying to solve or investigate this.

postal status shows all UP.
[root@server tmp]# postal status
      Name Command State Ports
-------------------------------------------------- ----------------
postal_cron_1 /docker-entrypoint.sh post ... Up
postal_requeuer_1 /docker-entrypoint.sh post ... Up
postal_web_1 /docker-entrypoint.sh post ... Up
postal_worker_1 /docker-entrypoint.sh post ... Up
postal_worker_10 /docker-entrypoint.sh post ... Up
postal_worker_11 /docker-entrypoint.sh post ... Up
postal_worker_12 /docker-entrypoint.sh post ... Up
postal_worker_13 /docker-entrypoint.sh post ... Up
postal_worker_14 /docker-entrypoint.sh post ... Up
postal_worker_15 /docker-entrypoint.sh post ... Up
postal_worker_16 /docker-entrypoint.sh post ... Up
postal_worker_17 /docker-entrypoint.sh post ... Up
postal_worker_18 /docker-entrypoint.sh post ... Up
postal_worker_19 /docker-entrypoint.sh post ... Up
postal_worker_2 /docker-entrypoint.sh post ... Up
postal_worker_20 /docker-entrypoint.sh post ... Up
postal_worker_3 /docker-entrypoint.sh post ... Up
postal_worker_4 /docker-entrypoint.sh post ... Up
postal_worker_5 /docker-entrypoint.sh post ... Up
postal_worker_6 /docker-entrypoint.sh post ... Up
postal_worker_7 /docker-entrypoint.sh post ... Up
postal_worker_8 /docker-entrypoint.sh post ... Up
postal_worker_9 /docker-entrypoint.sh post ... Up
[root@server tmp]#

**Comment by willpower232** [comment]
The release button implies the message is held not pending, is that correct?

It is possible for messages to be held because the credential you're using holds them or the email address you're trying to send it to has been added to the suppression list (in which case, releasing the message and it being successfully delivered usually clears it)

**Reply by julianorib** [reply]
Hello

the credentials are ok as it is used in other messages.

![2022-10-27 14_19_50-Window](https://user-images.githubusercontent.com/25228906/198357381-d79da5a6-7f4b-4ccf-b271-ac8e1951d42d.png)

and the recipient is not on the suppression list.

Here are some sample prints.

![1](https://user-images.githubusercontent.com/25228906/198357929-7bb175c3-0988-42c4-81d2-8fb052860e1d.png)
![2](https://user-images.githubusercontent.com/25228906/198357941-35d2bc7a-fd2f-4580-89eb-5ebb0d7b9eaf.png)
![3](https://user-images.githubusercontent.com/25228906/198357950-0245b36d-7aea-453c-84d5-4074f8ccb903.png)

the message is stopped as pending.
the message should be sent normally.

**Reply by willpower232** [reply]
ah that makes sense, your postal status from earlier seemed to be missing smtp, is that running somewhere else?

**Reply by julianorib** [reply]
yes. it is.

**Reply by julianorib** [reply]
[root@smtp-a ~]# postal status
    Name                   Command               State   Ports
--------------------------------------------------------------
postal_smtp_1   /docker-entrypoint.sh post ...   Up

**Reply by willpower232** [reply]
Good stuff, I guess you'll have to look at the database and see what the `queued_messages` table looks like. You should see the unsent messages in there and see if they've been `locked_by` anything.

**Reply by julianorib** [reply]
columns (locked_by and locked_at and retry_after) are set to NULL

**Reply by willpower232** [reply]
The requeuer is supposed to be picking them up, is it all messages or only some? Is there anything weird in its log?

**Reply by julianorib** [reply]
it's just a few messages.
in the log there is nothing apparently wrong.
Do you have any other suggestions to investigate?

**Reply by willpower232** [reply]
if its only a few messages then might be worth reading #2144 (and by extension #1190)

**Reply by julianorib** [reply]
Hello.

I checked the 2 topics and I believe I found the problem.
I use IP POOLs and share some POOLs between Organizations and other Servers. This could be what is causing the problem of some messages being pending.

I configured in another way, specific POOLs for each Server.
I'm testing.

Thanks.

**Comment by Ashishlead2** [comment]
Hi,
I am enjoying working with Postal, But also facing slow delivery(Pending). All 5 services are running with one worker. 

I have configured 1 organization and 16 Mail server. Tried to increase number of workers as per forum but can't see any increment. Can anyone please help

**Reply by willpower232** [reply]
have you tried using the `--scale` argument as per #1604 ?

**Reply by Ashishlead2** [reply]
> have you tried using the `--scale` argument as per #1604 ?

Yes I found this page after some time and It worked. Thanks, It's working well
